API.md

```diff
@@ -1,64 +1,272 @@
+  body: 'string'|'object', //The body to include in the outbound request. This overrides "data" if provided in method call.
+  mock: 'function',        //A mock runner to use. When not provided, uses "request" (i.e. does it for real).
+  form: 'boolean'          //Use the object in the body to create a form-encoded request.
+}
+```
+
+### get(uri, params)
+
+- Types: `uri`: `string`, `params`: `object` (optional)
+- Default: `none`
+
+Perform an HTTP GET on the specified URI.
+
+```javascript
+frisby.create('a test')
+    .get('http://example.com/login')
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+```
+
+### head(uri, params)
+
+Identical to [get](#geturi-params), using an HTTP HEAD request.
+
+### options(uri, params)
+
+Identical to [get](#geturi-params), using an HTTP OPTIONS request.
+
+### post(uri, data, params)
+
+- Types: `uri`: `string`, `data`:`object` `params`: `object` (optional)
+- Default: `none`
+
+Perform an HTTP POST to the specified URI.
+
+```javascript
+frisby.create('a test')
+    .post('http://example.com/login', {
+      username: 'joe@example.com',
+      password: 'J0£_£x@mpl£'
+    })
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+
+frisby.create('another test')
+    .post('http://example.com/contact-us', {
+      name: 'Test Person',
+      comment: 'What a lovely test'
+    }, {form: true})
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+
+```
+
+### put(uri, data, params)
+
+Identical to [post](#posturi-params), using an HTTP PUT request.
+
+### patch(uri, data, params)
+
+Identical to [post](#posturi-params), using an HTTP PATCH request.
+
+### delete(uri, data, params)
+
+Identical to [post](#posturi-params), using an HTTP DELETE request.
+
+### addHeader(header, content)
+
+Adds an HTTP header to your request
+
+- Types: `header`: `string`, `content`: `string`
+- Default: `none`
+
+Note that `content` is always a string, regardless of whether the data it would represent is something else (an integer or GUID for example).
+
+```javascript
+frisby.create('a test')
+    .get('http://example.com')
+    .addHeader('Accept', 'text/html')
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+```
+
+### addHeaders(headers)
+
+Adds a collection of headers to your request
+
+- Types: `headers`: `object`
+- Default: `none`
+
+A flat object where each key becomes a header name and each corresponding value becomes that header's value in the request.
+
+```javascript
+frisby.create('a test')
+    .get('http://example.com')
+    .addHeaders({
+      accept: 'text/html',
+      referer: 'http://www.test.net'
+    })
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+```
+
+### removeHeader(header)
+
+Removes a given header from the outgoing request
```

Comment:
It's not really clear here why this would be useful. Could you clarify that?

Reply:
I wouldn't like to pre-suppose what others might want to do with this functionality. The example I gave is where Content-Type is set by us, but the test developer doesn't want to send it, perhaps to test that their API is resilient to such things. I'd prefer to keep it in the docs unless we remove the function or rename it to private.

Reply:
I think it would be helpful to write some of what you just said into the docs! The docs should definitely tell a developer "what this does," but it's also really helpful to tell a developer "what can I do with this" / "why I might want to use this."

Reply:
Done!